core/src/main/scala/higherkindness/mu/rpc/srcgen/avro/AvroSrcGenerator.scala

```diff
@@ -166,22 +166,26 @@ final case class AvroSrcGenerator(
   }
 
   private def parseMessage(name: String, request: Schema, response: Schema): String = {
-    val args = request.getFields.asScala
-    if (args.size > 1)
+    val requestArgs = request.getFields.asScala
+    if (requestArgs.size > 1)
       throw ParseException("RPC method has more than 1 request parameter")
     val requestParam = {
-      if (args.isEmpty)
+      if (requestArgs.isEmpty)
         s"$DefaultRequestParamName: $EmptyType"
       else {
-        val arg = args.head
-        if (arg.schema.getType != Schema.Type.RECORD)
+        val requestArg = requestArgs.head
+        if (requestArg.schema.getType != Schema.Type.RECORD)
           throw ParseException("RPC method request parameter is not a record type")
-        s"${arg.name}: ${arg.schema.getFullName}"
+        s"${requestArg.name}: ${requestArg.schema.getFullName}"
       }
     }
     val responseParam = {
       if (response.getType == Schema.Type.NULL) EmptyType
-      else s"${response.getNamespace}.${response.getName}"
+      else {
+        if (response.getType != Schema.Type.RECORD)
+          throw ParseException("RPC method response parameter is not a record type")
```

Comment:
so I added an exception here to make sure that we only parse Avro schema with responses that have type `Record`.  But what I don't know how to do is to get this error to bubble up to the compiler so that we don't run into the error that's described in https://github.com/47deg/marlow/issues/513.  It's possible we don't need to have the error at the compiler level, but I don't know how to trip that condition before we try and compile the codebase that relies on source generation.

Reply:
Ope so I did some more digging on this and it looks like to make it so that this parsing errors at the compiler step it's likely that we'll have to make a change in `mu-scala`, not this plugin.  So, while this parsing is probably helpful, it's not needed to resolve issue [513](https://github.com/47deg/marlow/issues/513).  

@cb372 @naree any opinions on whether I should leave this in, or should I just close this PR and focus more on making the changes in `mu-scala`.